### ISSUE TITLE
Update Patreon provider URLs

### ISSUE THREAD

**Issue by evykassirer** [python-social-auth/social-core#307]
The path structure for Patreon URLs has changed recently as part of our update to V2.

- The access token URL has changed from `'https://api.patreon.com/oauth2/token``'` to `'https://www.patreon.com/api/oauth2/token``'`
- The revoke token URL has changed from `'https://api.patreon.com/oauth2/revoke``'` to `'https://www.patreon.com/oauth2/revoke'`
- The user data URL has changed from `'https://api.patreon.com/oauth2/api/current_user'` to `'https://www.patreon.com/api/oauth2/v2/identity``'` 

More information APIV2 here: https://docs.patreon.com/#apiv2-oauth

Please let me know if you have any questions!

**Comment by nstephenh** [comment]
I really wish I could get this working, but I can't figure out where I'd put credentials for tox or how I'd debug this. Will probably switch to a generic oauth implementation since it doesn't look like there have been any updates since September or so.

**Comment by rbricheno** [comment]
@nstephenh don't panic!

In addition to changing the base URL, v2 API requests must explicitly request the required fields by name. I found an example URL here:

https://www.patreondevelopers.com/t/patreon-apiv2-is-in-public-beta/1036/10

When I updated your PR #436 to use this URL, `tox` ran without error. See here for a diff:

https://github.com/python-social-auth/social-core/pull/436#issuecomment-595849180

I can't seem to raise this as a PR against your repo so you'll have to add it manually if you want your PR to pass tests.
